dispatcherd/service/metrics.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from typing import Any, Generator
+
+# Metrics library
+from prometheus_client import CollectorRegistry, make_asgi_app
+
+# For production of the metrics
+from prometheus_client.core import CounterMetricFamily
+from prometheus_client.metrics_core import Metric
+from prometheus_client.registry import Collector
+
+# General ASGI python web server, interfaces with prometheus-client lib by the ASGI standard
+from uvicorn.config import Config
+from uvicorn.server import Server
+
+from ..protocols import DispatcherMain
+
+logger = logging.getLogger(__name__)
+
+
+def metrics_data(dispatcher: DispatcherMain) -> Generator[Metric, Any, Any]:
+    """
+    Called each time metrics are gathered
+    This defines all the metrics collected and gets them from the dispatcher object
+    """
+    yield CounterMetricFamily(
+        'dispatcher_messages_received_total',
+        'Number of messages received by dispatchermain',
+        value=dispatcher.received_count,
+    )
+    yield CounterMetricFamily(
+        'dispatcher_control_messages_count',
+        'Number of control messages received.',
+        value=dispatcher.control_count,
+    )
+    yield CounterMetricFamily(
+        'dispatcher_worker_count',
+        'Number of workers running.',
+        value=len(list(dispatcher.pool.workers)),
+    )
+
+
+class CustomCollector(Collector):
+    def __init__(self, dispatcher: DispatcherMain) -> None:
+        self.dispatcher = dispatcher
+
+    def collect(self) -> Generator[Metric, Any, Any]:
+        for m in metrics_data(self.dispatcher):
+            yield m
+
+
+class DispatcherMetricsServer:
+    def __init__(self, port: int = 8070, log_level: str = 'info', host: str = "localhost") -> None:
```

Comment:
Sweet, you showed the `log_level` being passed through the config so the `port` and `host` can also be configured via the dispatcher yaml :+1: